app/models/product.rb

```diff
@@ -10,7 +11,26 @@ class Product
   field :version, type: String
   field :description, type: String
 
-  validates :name, presence: true
+  validates :name, presence: true, uniqueness: { scope: :vendor, message: "Product name was already taken. Please choose another." }
+  validates :ehr_type, presence: true
```

Comment:
Is this for EH/EP? If so, we should be validating those values, not just the presence